_includes/head.html

```diff
@@ -1,31 +1,44 @@
-  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+  <meta http-equiv="x-ua-compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no, viewport-fit=cover">
 
-  <link rel="shortcut icon" href="{{ site.url }}/assets/img/favicon.png">
   <title>{% unless page.url == '/' %}{{ page.title }} - {% endunless %}{{ site.title }}</title>
-  <meta name="description" content="{{ site.description }}">
-  <meta name="keywords" content="{{ site.title }}">
-
+  <meta name="description" content="{% if page.url == '/' %}{{ site.description }}{% elsif page.is_post %}{{ page.excerpt }}{% else %}{{ page.description }}{% endif %}">
+  <link rel="canonical" href="{{ site.url }}{{ page.url | replace:'index.html',''}}">
+  <link rel="shortcut icon" href="{{ site.url }}{{ site.baseurl }}/assets/img/favicon.png">
+  <link rel="apple-touch-icon" sizes="180x180" href="{{ site.url }}{{ site.baseurl }}/assets/img/apple-touch-icon.png">
+  <meta name="theme-color" content="#4C0704">
+  
+  <meta property="og:site_name" content="{{ site.title }}">
+  <meta property="og:title" content="{% if page.url == '/' %}{{ site.title }}{% else %}{{ page.title }}{% endif %}">
+  <meta property="og:description" content="{% if page.url == '/' %}{{ site.description }}{% elsif page.is_post %}{{ page.excerpt }}{% else %}{{ page.description }}{% endif %}">
+  <meta property="og:url" content="{{ site.url }}{{ page.url | replace:'index.html',''}}">
+  {% if page.is_post %}
+  <meta property="og:type" content="article">
+  <meta name="author" content="{{ page.author }}">
+  <meta property="article:author" content="{{ page.author }}">
+  <meta name="twitter:card" content="summary_large_image">
+  <meta property="og:image" content="{{ page.image }}">
+  {% else %}
   <meta property="og:type" content="website">
-  <meta property="og:title" content="{% unless page.url == '/' %}{{ page.title }} - {% endunless %}{{ site.title }}">
-  <meta property="og:description" content="{{ site.description }}">
-  <meta property="og:image" content="{{ site.url }}/assets/img/image-slider/12.jpg">
-  <meta property="og:url" content="{{ site.url }}{{ page.url }}">
+  <meta name="twitter:card" content="summary">
+  <meta property="og:image" content="{{ site.url }}{{ site.baseurl }}/assets/img/large-logo.png">
+  {% endif %}
+  <meta name="twitter:site" content="{{ '@' | append: site.data.links.twitter | replace: 'https://twitter.com/', '' }}">
 
-  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css" integrity="sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO" crossorigin="anonymous" />
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/4.6.2/css/bootstrap.min.css" integrity="sha384-xOolHFLEh07PJGoPkLv1IbcEPTNtaed2xpHsD9ESMhqIYd0nLMwNLD69Npy4HI+N" crossorigin="anonymous">
```

Comment:
Curious, were there any major differences between the bootstrap versions? (Smart move, just curious what changed)

Reply:
It's essentially just quality of life and bug fixes, some which might be necessary to contributors that edit the website or the users visiting https://oc.tc (think better support for accessibility settings, etc.) Without going into more details, you can read [Bootstrap's changelogs starting from 4.1.3](https://github.com/twbs/bootstrap/releases?page=4). I was originally going to upgrade to Bootstrap 5, however, it would've required rewriting a majority of the website and it would drop support for Internet Explorer and older version of browsers.

Reply:
Thanks! Good to know.